src/runtime_src/core/common/drv/kds_cu_ctrl.c

```diff
@@ -0,0 +1,294 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Xilinx Kernel Driver Scheduler
+ *
+ * Copyright (C) 2020 Xilinx, Inc.
+ *
+ * Authors: min.ma@xilinx.com
+ */
+
+#include "kds_cu_ctrl.h"
+
+static int
+get_cu_by_addr(struct kds_cu_ctrl *kcuc, u32 addr)
+{
+	int i;
+
+	/* Do not use this search in critical path */
+	for (i = 0; i < kcuc->num_cus; ++i) {
+		if (kcuc->xcus[i]->info.addr == addr)
+			break;
+	}
+
+	return i;
+}
```

Comment:
It is a little bit confusing here. What if we could not find the cu (idx?) by given address? how about
static int get_cu_idx_by_addr (struct kds_cu_ctrl *kcuc, u32 addr, int *cu_idx)
?

Reply:
If CU is not found, it would return num_cus, which is out of range.
If this function would return any other error type, we could change this. Let me add a command for this function.

BTW, at the end of the day, once all of the modules become stable, we should commend all functions.